src/game/scenes/mainmenu/menu_advanced.c

```diff
@@ -0,0 +1,94 @@
+#include "game/scenes/mainmenu/menu_advanced.h"
+
+#include "game/common_defines.h"
+#include "game/gui/gui.h"
+#include "game/utils/settings.h"
+
+#include "resources/languages.h"
+#include "utils/miscmath.h"
+
+typedef struct {
+    int throw_range;
+    int jump_height;
+    int vitality;
+    int block_damage;
+} menu_advanced_local;
+
+void menu_advanced_done(component *c, void *u) {
+    menu *m = sizer_get_obj(c->parent);
+    menu_advanced_local *local = menu_get_userdata(c->parent);
+
+    // convert the positions back into values
+    settings_get()->advanced.throw_range = local->throw_range * 20;
+    settings_get()->advanced.jump_height = (local->jump_height * 10) + 80;
+    settings_get()->advanced.vitality = (local->vitality * 20) + 80;
+    settings_get()->advanced.block_damage = local->block_damage * 5;
+
+    omf_free(local);
+
+    m->finished = 1;
+}
+
+component *menu_advanced_create(scene *s) {
+    component *menu = menu_create();
+    menu_advanced_local *local = omf_calloc(1, sizeof(menu_advanced_local));
+    menu_set_userdata(menu, local);
+
+    const char *on_off_opts[] = {"OFF", "ON"};
+    const char *throw_ranges[] = {"0%",   "20%",  "40%",  "60%",  "80%",  "100%", "120%", "140%",
+                                  "160%", "180%", "200%", "220%", "240%", "260%", "280%", "300%"};
+    const char *jump_heights[] = {"80%", "90%", "100%", "110%", "120%", "130%"};
+    const char *hit_pauses[] = {"0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10"};
+    const char *vitalities[] = {"80%",  "100%", "120%", "140%", "160%", "180%", "200%", "220%", "240%",
+                                "260%", "280%", "300%", "320%", "340%", "360%", "380%", "400%"};
+    const char *knock_downs[] = {"NONE", "KICKS", "PUNCHES", "BOTH"};
+    const char *block_damages[] = {"0%", "5%", "10%", "15%", "20%", "25%", "30%", "35%"};
+
+    // convert the values from config into indexes into the option lists
+    local->throw_range = clamp(settings_get()->advanced.throw_range / 20, 0, 15);
+    local->jump_height = clamp((settings_get()->advanced.jump_height - 80) / 10, 0, 5);
+    local->vitality = clamp((settings_get()->advanced.vitality - 80) / 20, 0, 16);
+    local->block_damage = clamp(settings_get()->advanced.block_damage / 5, 0, 7);
+    menu_attach(menu, label_create_title("ADVANCED"));
+    menu_attach(menu, filler_create());
+
+    component *c = textselector_create_bind_opts("REHIT MODE", lang_get(276), NULL, NULL,
+                                                 &settings_get()->advanced.rehit_mode, on_off_opts, 2);
+    component_disable(c, 1);
+    menu_attach(menu, c);
+
+    c = textselector_create_bind_opts("DEF. THROWS", lang_get(277), NULL, NULL,
+                                      &settings_get()->advanced.defensive_throws, on_off_opts, 2);
+
+    component_disable(c, 1);
+    menu_attach(menu, c);
+    c = textselector_create_bind_opts("THROW RANGE", lang_get(278), NULL, NULL, &local->throw_range, throw_ranges, 16);
```

Comment:
could we use the language enums for the new lang_get calls ? I think we have the enum now ?

Reply:
For the option names, not just the descriptions?

Reply:
```ID: 261
Title: Option 3
Data: THROW RANGE %i%%a```

They're a bit inconvienient to use with our existing widgets...

Reply:
places where we have the lang_get() in use now with a plain integer.

Reply:
oh, I see

Reply:
Can you point me at the enum you want me to use?

Reply:
no, never mind, looks like it was not yet merged; it is sitting in nopeys translations branch :/